packages/amplify-graphql-default-value-transformer/src/graphql-default-value-transformer.ts

```diff
@@ -72,10 +72,21 @@ const validateDefaultValueType = (ctx: TransformerSchemaVisitStepContextProvider
   }
 };
 
+const validateNotPrimaryKey = (field: FieldDefinitionNode): void => {
+  const isPrimaryKeyField =
+    field.directives!.find((dir) => dir.name.value === 'primaryKey') ||
+    (getBaseType(field.type) === 'ID' && field.type.kind === Kind.NON_NULL_TYPE && field.name.value === 'id');
+
```

Comment:
There is one more case to consider here which is when the `@default` is applied to fields part of a composite primary key. [Gen2 doc](https://docs.amplify.aws/react/build-a-backend/data/data-modeling/identifiers/#composite-identifier) or this [Gen1 doc is more explicit](https://docs.amplify.aws/gen1/react/build-a-backend/graphqlapi/data-modeling/#configure-a-primary-key) about the schema with composite primary key.

Reply:
Added a separate check scanning over `sortKeyFields`.